plugin/src/main/java/org/jetbrains/research/refactorinsight/actions/HideNonFunctionalChangesAction.java

```diff
@@ -0,0 +1,70 @@
+package org.jetbrains.research.refactorinsight.actions;
+
+import com.intellij.diff.contents.DiffContent;
+import com.intellij.diff.editor.DiffVirtualFile;
+import com.intellij.diff.editor.SimpleDiffVirtualFile;
+import com.intellij.diff.requests.DiffRequest;
+import com.intellij.diff.requests.SimpleDiffRequest;
+import com.intellij.diff.tools.util.DiffDataKeys;
+import com.intellij.diff.tools.util.base.TextDiffSettingsHolder;
+import com.intellij.diff.tools.util.text.SimpleTextDiffProvider;
+import com.intellij.openapi.Disposable;
+import com.intellij.openapi.actionSystem.AnActionEvent;
+import com.intellij.openapi.actionSystem.CommonDataKeys;
+import com.intellij.openapi.editor.Editor;
+import com.intellij.openapi.editor.FoldRegion;
+import com.intellij.openapi.project.DumbAwareToggleAction;
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.util.Disposer;
+import com.intellij.openapi.vfs.VirtualFile;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.List;
+
+public class HideNonFunctionalChangesAction extends DumbAwareToggleAction {
```

Comment:
I would suggest to make it as a [checkbox](https://jetbrains.design/intellij/controls/checkbox/) with a text "Hide non-functional changes", not a button.

Reply:
Do you mean a checkbox for a list of items or just for one - hide/not hide?

Reply:
One checkbox for hide/not hide